packages/segmentedControl/components/SegmentedControlButton.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from "react";
+import { cx } from "emotion";
+import shortid from "shortid";
+import { segmentedControlButton, segmentedControlButtonActive } from "../style";
+import { visuallyHidden, display } from "../../shared/styles/styleUtils";
+
+export interface SegmentedControlButtonProps {
+  /**
+   * Unique identifier used for the input button
+   */
+  id?: string;
+  /**
+   * Whether the button has been selected
+   */
+  isActive?: boolean;
+  /**
+   * The name passed to the input. Shared by all inputs in the segmented control
+   */
+  name?: string;
+  /**
+   * Callback for when a user makes a selection. Use the `onSelect` prop on the `SegmentedControl` instead, unless there is some extenuating circumstance.
+   */
+  onChange?: (e: React.SyntheticEvent<HTMLElement>) => void;
+  /**
+   * The value of the input
+   */
+  value: string;
+}
+
+const SegmentedControlButton: React.SFC<
+  SegmentedControlButtonProps
+> = props => {
+  const { id = shortid.generate(), isActive, onChange, name, value } = props;
```

Comment:
not that i see that as a really problematic thing, but on every change of the selection, we'll generate a new `shortid` for all the buttons within a `SegmentedControl` because of the `.bind` we use in there.

it somehow feels a little odd to me but i can't think of critical arguments against that behaviour (it's only recreationg of 5 DOM-nodes at max if used properly. hooray for the documentation btw!).